pandas/core/algorithms.py

```diff
@@ -717,6 +717,17 @@ def select_n(series, n, keep, method):
     return dropped.iloc[inds]
 
 
+def select_n_frame(frame, columns, n, method, keep):
+    from pandas.core.series import Series
+    if not is_list_like(columns):
```

Comment:
thanks for moving. If you are interested in next PR! would want to clean up this code w.r.t. largest/smallest a bit more anyhow; IOW, make it cleaner / nicer. (and I think we have a perf issue outstanding I think with Series).